### PR TITLE
[Model] - Create Product Model

### PR DESCRIPTION
**Pull request by vinibr3**
Created product model and his tests.

### CODE REVIEW

File: spec/models/service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe Service, type: :model do
+  it { should validate_numericality_of(:code).only_integer }
+  it { should validate_numericality_of(:code).is_greater_than(0) }
+  it { should validate_presence_of(:code) }
+  it { should validate_length_of(:code).is_at_most(255) }
+  it "Uniqueness of code" do
+    create(:product)
+    validate_uniqueness_of(:code)
```
Comment (kaiomagalhaes) [inline spec/models/service_spec.rb:10]: where is the expectation?

File: spec/models/service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe Service, type: :model do
+  it { should validate_numericality_of(:code).only_integer }
```
Comment (kaiomagalhaes) [inline spec/models/service_spec.rb:4]: Please use `expect` instead of `should`

File: spec/models/service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  it "Uniqueness of code" do
+    create(:product)
+    validate_uniqueness_of(:code)
+  end
+
+  it { should validate_length_of(:name).is_at_most(255) }
+  it { should validate_presence_of(:name) }
+  it "Uniqueness of name" do
+    create(:product)
+    validate_uniqueness_of(:name)
```
Comment (kaiomagalhaes) [inline spec/models/service_spec.rb:17]: where is the expectation?

File: spec/models/service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  it { should validate_presence_of(:description) }
```
Comment (kaiomagalhaes) [inline spec/models/service_spec.rb:20]: Please use expect instead of should

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: spec/factories/service.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :service do
+    digit = Faker::Number.between(1, 10)
+
+    code { Faker::Number.number(digit) }
+    name { Faker::Name.name }
+    description { Faker::Lorem.paragraph }
+    value { Faker::Number.decimal(digit, 2) }
```
Comment (RodrigoNM) [inline spec/factories/service.rb:8]: It's not neccessary to do this, you don't need line 3.
On line 8 you can just do Faker::Number.positive.round(2)

File: spec/models/installment_spec.rb [hunk 6]
```diff
@@ -10,6 +10,7 @@
   it { is_expected.to validate_presence_of(:payment_date) }
   it { is_expected.to validate_numericality_of(:total_value).is_greater_than_or_equal_to(0) }
   it { is_expected.to validate_presence_of(:total_value) }
+  it { should belong_to(:payment_method).validate(true) }
```
Comment (RodrigoNM) [inline spec/models/installment_spec.rb:13]: it's better to use expect

**Pull request closed without merging**